keras_cv/layers/preprocessing/rgb_shift.py

```diff
@@ -0,0 +1,144 @@
+    rgbshift = keras_cv.layers.preprocessing.RGBShift()
+    augmented_images = rgbshift(images)
+    ```
+    """
+
+    def __init__(
+        self,
+        r_shift_limit=10,
+        g_shift_limit=10,
+        b_shift_limit=10,
+        seed=None,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self._r_shift_limit = self._set_shift_limit(r_shift_limit, channel='r_shift_limit')
+        self._g_shift_limit = self._set_shift_limit(g_shift_limit, channel='g_shift_limit')
+        self._b_shift_limit = self._set_shift_limit(b_shift_limit, channel='b_shift_limit')
+        self.seed = seed
+
+    def _set_shift_limit(self, shift_limit, channel=''):
+        if isinstance(shift_limit, (tuple, list)):
+            if len(shift_limit) != 2: 
+                raise ValueError(f'The {channel} should be scalar, tuple or list of two upper and lower \
+                    bound number. Got {shift_limit}')
+            return sorted(shift_limit)
+        elif isinstance(shift_limit, (int, float)):
+            shift_limit = abs(shift_limit)
+            return [-shift_limit, shift_limit]
+        else:
+            raise ValueError(f'The {channel} should be scalar, tuple or list of two upper and lower bound\
+                 number. Got {shift_limit}')
+
+    @tf.function
+    def _rgb_shifting(self, image):
+        r_shift = tf.random.uniform([], 
+                                    minval=self._r_shift_limit[0], 
+                                    maxval=self._r_shift_limit[1], 
+                                    dtype=tf.float32, seed=self.seed)
+        g_shift = tf.random.uniform([], 
+                                    minval=self._g_shift_limit[0], 
+                                    maxval=self._g_shift_limit[1], 
+                                    dtype=tf.float32, seed=self.seed)
+        b_shift = tf.random.uniform([], 
+                                    minval=self._b_shift_limit[0],
+                                    maxval=self._b_shift_limit[1], 
+                                    dtype=tf.float32, seed=self.seed)
+        
+        unstack_rgb = tf.unstack(image, axis=-1)
+        shifted_rgb = tf.stack([unstack_rgb[0] + r_shift, 
+                                unstack_rgb[1] + g_shift, 
+                                unstack_rgb[2] + b_shift],  axis=-1)
+        shifted_rgb = tf.clip_by_value(shifted_rgb, 0, 255)
+        return tf.cast(shifted_rgb, image.dtype)
+
+    def call(self, images, training=None):
+        """call method for the RGBShift layer.
+
+        Args:
+            images: Tensor representing images of shape
+                [batch_size, width, height, channels], with dtype tf.float32 / tf.uint8, or,
+                [width, height, channels], with dtype tf.float32 / tf.uint8
+        Returns:
+            images: augmented images, same shape as input.
+        """
+        if training is None:
+            training = backend.learning_phase()
+
+        if not training:
```

Comment:
Please use the tf.cond here since the `training` value could be a tensor, which will fail with the if check.

Reply:
Done.